File: 02_week/06_day/cpp/main.cpp
#include <iostream>
#include <random>
using namespace std;

void generate_random_pwd(char *pwd, int len) {

	random_device rd;
	mt19937 mt(rd());
	for (int i = 0; i < len; i++) {
		int rand_select = mt() % 4 + 1;

		if (rand_select == 1)
			pwd[i] = mt() % 26 + 'a';
		else if(rand_select == 2)
			pwd[i] = mt() % 26 + 'A';
		else if(rand_select == 3)
			pwd[i] = mt() % 10 + '0';
		else
			pwd[i] = mt() % 15 + 33;
	}
}

void print_pwd(const char *pwd, int len) {
	for (int i = 0; i < len; i++)
		cout << pwd[i];
	cout << '\n';
}

int main() {
	const int len = 10;
	char pwd[len];

	cout << "랜덤 10자리 비밀번호 생성기" << '\n';

	generate_random_pwd(pwd, len);
	print_pwd(pwd, len);

	return 0;
}

File: 02_week/06_day/rust/main.rs
use rand::Rng;

fn main() {

    println!("랜덤 10자리 비밀번호 생성기");

    let pwd = generate_random_pwd();

    for ch in pwd {
        print!("{}", ch);
    }

    println!();
}

fn generate_random_pwd() -> [char; 10] {

    let len: usize = 10;
    let mut pwd: [char; 10] = ['_'; 10];

    let mut i: usize = 0;
    while i < len {
        let rand_select = rand::thread_rng().gen_range(1..=4);

        let rand_pwd = if rand_select == 1 {
            rand::thread_rng().gen_range(97..=122) as u8 as char  // 특수문자
        }
        else if rand_select == 2 {
            rand::thread_rng().gen_range(65..=90) as u8 as char  // 대문자
        }
        else if rand_select == 3 {
            rand::thread_rng().gen_range(48..=57) as u8 as char  // 숫자
        }
        else {
            rand::thread_rng().gen_range(33..=47) as u8 as char  // 소문자
        };

        pwd[i] = rand_pwd;
        i += 1;
    }

    pwd
}